code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapIdentitySearchProvider.java

```diff
@@ -368,28 +368,27 @@ private Identity attributesToIdentity(Attributes search){
             if (!hasPermission(search)){
                 throw new ClientVisibleException(ResponseCodes.UNAUTHORIZED);
             }
+            Attribute nameField;
             String externalIdType;
             String accountName;
             String externalId = (String) search.get(LdapConstants.DN).get();
             String login;
             if (isType(search, LdapConstants.USER_OBJECT_CLASS.get())){
                 externalIdType = LdapConstants.USER_SCOPE;
-                if (search.get(LdapConstants.USER_NAME_FIELD.get()) != null) {
-                    accountName = (String) search.get(LdapConstants.USER_NAME_FIELD.get()).get();
-                } else {
-                    accountName = externalId;
-                }
-
+                nameField = search.get(LdapConstants.USER_NAME_FIELD.get());
             } else if (isType(search, LdapConstants.GROUP_OBJECT_CLASS.get())) {
                 externalIdType = LdapConstants.GROUP_SCOPE;
-                if (search.get(LdapConstants.GROUP_NAME_FIELD.get()) != null) {
-                    accountName = (String) search.get(LdapConstants.GROUP_NAME_FIELD.get()).get();
-                } else {
-                    accountName = externalId;
-                }
+                nameField = search.get(LdapConstants.GROUP_NAME_FIELD.get());
             } else {
                 return null;
             }
+            if (nameField != null) {
```

Comment:
@cjellick @hibooboo2 I've meant that we should put "non empty" check upon retrieving the value from the attribute via (get) by a) casting it to String b) checking for non-empty. Otherwise empty value assignment is possible on lines 386/388.